operator/postgresqlcontroller/observe.go

```diff
@@ -35,7 +35,7 @@ func (p *Pipeline) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 	isUpToDate := mpr.IsResourceUpToDate(pgInstance, resp.JSON200)
 	setConditionFromState(pgExo, pgInstance)
 
-	ca, err := p.exo.GetDatabaseCACertificate(ctx, pgInstance.Spec.ForProvider.Zone)
+	ca, err := p.exo.GetDatabaseCACertificate(ctx, string(pgInstance.Spec.ForProvider.Zone))
```

Comment:
I suggest to add `func (z Zone) String() string` to the Zone, so that it implements fmt.Stringer and we can do this instead:
```suggestion
	ca, err := p.exo.GetDatabaseCACertificate(ctx, pgInstance.Spec.ForProvider.Zone.String())
```